onmetal-controller-manager/controllers/networking/natgateway_controller_test.go

```diff
@@ -0,0 +1,278 @@
+			Expect(client.IgnoreNotFound(err)).NotTo(HaveOccurred())
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(metav1.IsControlledBy(natGatewayRouting, natGateway)).To(BeTrue(), "nat gateway routing is not controlled by nat gateway: %#v", natGatewayRouting.OwnerReferences)
+			g.Expect(natGatewayRouting.Destinations).To(BeEmpty())
+		}).Should(Succeed())
+
+		natGatewayBase := natGateway.DeepCopy()
+		ip := netip.IPv4Unspecified()
+		for _, v := range natGateway.Spec.IPs {
+			ip = ip.Next()
+			natGateway.Status.IPs = append(natGateway.Status.IPs, networkingv1alpha1.NATGatewayIPStatus{
+				Name: v.Name,
+				IP: commonv1alpha1.IP{
+					Addr: ip,
+				},
+			})
+		}
+		Expect(k8sClient.Patch(ctx, natGateway, client.MergeFrom(natGatewayBase))).To(Succeed())
+
+		By("creating a network interfaces")
+
+		nics := 64
+		for i := 0; i < nics; i++ {
+			nic := &networkingv1alpha1.NetworkInterface{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    ns.Name,
+					GenerateName: "nic-",
+					Labels:       map[string]string{"foo": "bar"},
+				},
+				Spec: networkingv1alpha1.NetworkInterfaceSpec{
+					NetworkRef: corev1.LocalObjectReference{Name: network.Name},
+					IPFamilies: []corev1.IPFamily{
+						corev1.IPv4Protocol,
+					},
+					IPs: []networkingv1alpha1.IPSource{
+						{
+							Value: commonv1alpha1.MustParseNewIP("10.0.0.1"),
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, nic)).To(Succeed())
+		}
+
+		By("waiting for the nat gateway routing to be updated")
+		Eventually(func(g Gomega) {
+			Expect(k8sClient.Get(ctx, natGatewayKey, natGatewayRouting)).To(Succeed())
+			g.Expect(natGatewayRouting.Destinations).To(HaveLen(nics))
+		}).Should(Succeed())
+
+		for i := 0; i < nics; i++ {
+			nic := &networkingv1alpha1.NetworkInterface{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    ns.Name,
+					GenerateName: "nic-",
+					Labels:       map[string]string{"foo": "bar"},
+				},
+				Spec: networkingv1alpha1.NetworkInterfaceSpec{
+					NetworkRef: corev1.LocalObjectReference{Name: network.Name},
+					IPFamilies: []corev1.IPFamily{
+						corev1.IPv4Protocol,
+					},
+					IPs: []networkingv1alpha1.IPSource{
+						{
+							Value: commonv1alpha1.MustParseNewIP("10.0.0.1"),
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, nic)).To(Succeed())
+		}
+
+		totalSlots := int(((MAX_PORT-MIN_PORT)/portsPerNetworkInterface)+1) * 2
+		By("waiting for the nat gateway routing to be updated")
+		Eventually(func(g Gomega) {
+			Expect(k8sClient.Get(ctx, natGatewayKey, natGatewayRouting)).To(Succeed())
+			g.Expect(natGatewayRouting.Destinations).To(HaveLen(2 * nics))
+			var assigned, unassigned int
+			for _, v := range natGatewayRouting.Destinations {
+				if len(v.IPs) == 0 {
+					unassigned++
+				} else {
+					assigned++
+				}
+			}
+			g.Expect(assigned).To(BeEquivalentTo(totalSlots))
+			g.Expect(unassigned).To(BeEquivalentTo(nics*2 - totalSlots))
```

Comment:
We should verify here that the first set of `NetworkInterface`s is present in the `Destination`s while the second set of `NetworkInterface`s is not present here.